accord-core/src/main/java/accord/coordinate/FetchData.java

```diff
@@ -347,61 +353,84 @@ public Void apply(SafeCommandStore safeStore)
         {
             SafeCommand safeCommand = safeStore.get(txnId, this, route);
             Command command = safeCommand.current();
-            if (command.saveStatus().phase.compareTo(Phase.Persist) >= 0)
-                return null;
 
-            Status propagate = achieved.propagate();
+            PartialTxn partialTxn = this.partialTxn;
+            PartialDeps partialDeps = this.partialDeps;
+            switch (command.saveStatus().phase)
+            {
+                case Persist: return updateDurability(safeStore, safeCommand);
+                case Cleanup: return null;
+            }
+
+            Known achieved = this.achieved;
+            if (isTruncated)
+            {
+                achieved = applyOrUpgradeTruncated(safeStore, safeCommand, command);
+                if (achieved == null)
+                    return null;
+
+                if (achieved.executeAt.isDecided())
+                {
+                    Timestamp executeAt = command.executeAtIfKnown(full.executeAt);
+                    if (partialTxn == null && this.full.saveStatus.known.definition.isKnown())
```

Comment:
The idea here is to permit partial application on the node when the transaction has been partially cleaned up elsewhere in the cluster, so that the node has insufficient information to apply across all of its shards. `propagate` computes a _whole node_ propagation, but the individual shards are permitted to upgrade to applying either only to themselves, or to part of themselves (if they determine they are stale).